tests/e2e/default/reports/sms-messages.wdio-spec.js

```diff
@@ -0,0 +1,80 @@
+const utils = require('@utils');
+const commonElements = require('@page-objects/default/common/common.wdio.page');
+const reportsPage = require('@page-objects/default/reports/reports.wdio.page');
+const loginPage = require('@page-objects/default/login/login.wdio.page');
+const placeFactory = require('@factories/cht/contacts/place');
+const personFactory = require('@factories/cht/contacts/person');
+const reportFactory = require('@factories/cht/reports/generic-report');
+
+describe('Reports tab messages', () => {
+  const places = placeFactory.generateHierarchy();
+  const clinic = places.get('clinic');
+  const health_center = places.get('health_center');
+  const district_hospital = places.get('district_hospital');
+  const contact = {
+    _id: 'fixture:user:user1',
+    name: 'OfflineUser',
+    phone: '+12068881234',
+    place: health_center._id,
+    type: 'person',
+    parent: {
+      _id: health_center._id,
+      parent: health_center.parent
+    },
+  };
+
+  const patient = personFactory.build({
+    _id: 'patient_uuid',
+    parent: { _id: clinic._id, parent: { _id: health_center._id, parent: { _id: district_hospital._id }}},
+    name: 'the_patient',
+    patient_id: 'the_patient_id'
+  });
+
+  const reports = [
+    reportFactory
+      .report()
+      .build(
+        {
+          form: 'P',
+          content_type: 'xml',
+          scheduled_tasks: [{
+            due: '2024-05-09T04:30:00.000Z',
+            group: 8,
+            type: 'ANC Reminders LMP',
+            translation_key: 'instance.upgrade.version',
+            message_key: 'messages.c.report_accepted',
+            recipient: 'clinic',
+            state_history: [{
+              state: 'scheduled',
+              timestamp: '2024-02-26T10:42:41.080Z'
+            }],
+            state: 'scheduled'
+          },]
+        },
+        {
+          patient: { _id: patient._id, patient_id: '' },
+          submitter: contact,
+          fields: { lmp_date: 'Feb 3, 2022', patient_name: patient.name },
+        },
+      ),
```

Comment:
Can we replicate this scenario by following each step of the workflow with the web app, from patient creation and report generation?

Reply:
It would be a little tricky to simulate the exact scenario, besides needing a lot of setup with config, it would also require editing some dates, to make sure that we simulate the conditions exactly. 

Can you explain the scenario you're requesting in detail and what you expect to see tested?

Reply:
My idea was to do the entire workflow: Create the patient, send the SMS report P, and verify. But if editing dates are required, how you implemented is better. Thanks.

Reply:
This is what I'm thinking the test would require: 
- update settings to add the schedule to one of the existent forms, enable `registration` transition for the form to add the schedule. the schedule should have a message that would be sent sometime in the future, but more than 7 days into the future
- enable `generate_shortcode_on_contacts` transition
- create an offline user
- login as the offline user
- while offline, create a patient and then create a report for that patient
- go online and sync 
- wait for Sentinel to process both docs
- sync
- check the report in webapp and validate that webapp generates the message correctly
- if we want to check the sentinel message generating, we would need to edit the report manually, to set the due date of the scheduled task into the near future and restart sentinel, wait for due tasks to process, sync and run the same assertions again after the state has changed to `pending`.

Let me know if you think this is worth writing.

Reply:
Thanks, @dianabarsan, for putting together the test steps. That was what I was thinking, and now I have a clearer idea of how this works. 
If the first two steps are complicated, that can be an issue. But the last step worries me, and it worries me for a couple of things. First, editing the report manually is almost the same as creating and saving the [report](https://github.com/medic/cht-core/blob/b90e34d60fd4091792b5bcc8fa8c28ef48dce3bb/tests/e2e/default/reports/sms-messages.wdio-spec.js#L33) with the necessary values. Second, and most importantly, restarting, waiting, and syncing can easily introduce flakiness. What do you think?

Reply:
I think you're right, that the setup is really hard to achieve without it being flaky or requiring manual intervention. 
This was why I was reluctant to add this test.